### PR TITLE
Fix #333

### LINKED ISSUE #333: Logging in with Google no longer works
When attempting to log in, the user gets redirected to this page:

https://support.google.com/accounts/answer/6206245?p=openid&rd=1

### PR DESCRIPTION
**Pull request by adrianbanks**
This fixes #333.

This is a bit of a pig. I've fixed the logging in behaviour, but there is an outstanding issue that I can't fix. Any user who has already logged in via Google, when logging in again will get a new account. There isn't an upgrade path for these logins.

To mitigate it, I've added a warning message asking people to contact us. I think that is pants however.

Anyone got any better ideas?

(There is supposedly an upgrade path, but I can't get through the confusion of the docs to figure out how to make it work correctly.)

### CODE REVIEW

**Comment by alastairs** [comment]
I've not looked at the code yet, but I think we may be able to mitigate this a little. Can we chat about it before the end of the day? I'm keen to get this fixed before submissions open if possible.

**Comment by adrianbanks** [comment]
Sure. I'm open to any ideas.

**Comment by alastairs** [comment]
Can't be merged until the build is fixed - there's some kind of NuGet error:

"The build restored NuGet packages. Build the project again to include these packages in the build. For more information, see http://go.microsoft.com/fwlink/?LinkID=317568"

File: DDDEastAnglia/DDDEastAnglia.csproj [hunk 1]
```diff
@@ -747,4 +757,9 @@
     </PropertyGroup>
     <Error Condition="!Exists('$(SolutionDir)\.nuget\NuGet.targets')" Text="$([System.String]::Format('$(ErrorText)', '$(SolutionDir)\.nuget\NuGet.targets'))" />
   </Target>
+  <Import Project="..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets" Condition="Exists('..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets')" />
```
Comment (alastairs) [inline DDDEastAnglia/DDDEastAnglia.csproj:761]: Perhaps it's this that caused the build error?

File: DDDEastAnglia/Views/Account/ExternalLoginConfirmation.cshtml [hunk 2]
```diff
@@ -12,6 +12,33 @@
     <fieldset>
         <legend>Associate your @ViewBag.ProviderDisplayName account</legend>
 
+        @if (string.Equals("google", ViewBag.ProviderDisplayName, StringComparison.InvariantCultureIgnoreCase))
```
Comment (alastairs) [inline DDDEastAnglia/Views/Account/ExternalLoginConfirmation.cshtml]: Could this be factored out into a partial view? I don't mind whether or not that partial contains the `if` statement too -whichever you think best.

**Comment by alastairs** [comment]
Wow, DotNetOpenAuth _really_ likes screwing with your Web.config...

:+1: for the changes beyond the one pretty minor refactoring.

**Pull request merged**